app/views/contexts/preview/preview_question.py

```diff
@@ -0,0 +1,92 @@
+class PreviewQuestion:
+    def __init__(
+        self,
+        question_schema,
+    ):
+        self.title = question_schema.get("title")
+        self.answers = self._build_answers(question_schema)
+        self.descriptions = self._build_descriptions(question_schema)
+        self.guidance = self._build_question_guidance(question_schema)
+        self.text_length = self._get_length(question_schema.get("answers"))
+        self.instruction = question_schema.get("instruction") or None
+        self.answer_description = self._build_answer_descriptions(
+            iter(question_schema["answers"])
+        )
+        self.answer_guidance = self._build_answer_guidance(
+            answers=iter(question_schema["answers"])
+        )
+
+    @staticmethod
+    def _build_answers(question_schema):
+        answers = []
+        for answer in iter(question_schema["answers"]):
+            if options := answer.get("options"):
+                answers.extend(option["label"] for option in options)
+            if not options:
+                answers.append(answer["label"])
+
+        return answers
+
+    @staticmethod
+    def _build_answer_descriptions(answers):
+        return next(
+            (
+                answer.get("description")
+                for answer in answers
+                if answer.get("description")
+            ),
+            None,
+        )
+
+    def _build_answer_guidance(self, answers):
+        for answer in answers:
+            return self._build_guidance(answer)
+
+    @staticmethod
+    def _build_descriptions(question_schema):
+        if description := question_schema.get("description"):
```

Comment:
I think we need to handle placeholders in places like question descriptions as it looks this would fall over at the moment. Maybe we need to update the schema with examples of placeholders in a description etc

Reply:
Updated the schema (added new descriptions to a question), tests, resolving placeholders in question description now (using metadata if possible). Thanks.